www/js/lib/zimArchive.js

```diff
@@ -46,17 +46,38 @@ define(['zimfile', 'zimDirEntry', 'util', 'utf8'],
      * Creates a ZIM archive object to access the ZIM file at the given path in the given storage.
      * This constructor can also be used with a single File parameter.
      * 
-     * @param {StorageFirefoxOS|Array.<Blob>} storage Storage (in this case, the path must be given) or Array of Files (path parameter must be omitted)
-     * @param {String} path
-     * @param {callbackZIMArchive} callbackReady
+     * @param {StorageFirefoxOS|Array<Blob>} storage Storage (in this case, the path must be given) or Array of Files (path parameter must be omitted)
+     * @param {String} path The Storage path for an OS that requires this to be specified
+     * @param {callbackZIMArchive} callbackReady The function to call when the archive is ready to use
      */
     function ZIMArchive(storage, path, callbackReady) {
         var that = this;
         that._file = null;
         that._language = ""; //@TODO
-        var createZimfile = function(fileArray) {
-            zimfile.fromFileArray(fileArray).then(function(file) {
+        var createZimfile = function (fileArray) {
+            zimfile.fromFileArray(fileArray).then(function (file) {
                 that._file = file;
+                // File has been created, but we need to add any Listings which extend the archive metadata
+                that._file.setListings([
+                    // Provide here any Listings for which we need to extract metadata as key:value obects to be added to the file
+                    // 'ptrName' and 'countName' contain the key names to be set in the archive file object
+                    {
+                        path: 'X/listing/titleOrdered/v0',
+                        ptrName: 'titlePtrPos',
+                        countName: 'entryCount'
+                    },
+                    {
+                        // !!! CHANGE THIS PATH TO .../v1 BEFORE MERGING !!!
+                        // It is set to v0 for testing because we do not have any archive with a valid v1 index yet
+                        path: 'X/listing/titleOrdered/v0',
```

Comment:
This is where the code is rigged. `v0` needs to be changed to `v1` as soon as we have a ZIM with a valid, non-empty v1 title pointer list.